exporter/elasticsearchexporter/README.md

```diff
@@ -83,32 +83,52 @@ The Elasticsearch exporter supports the common [`sending_queue` settings][export
 ### Elasticsearch document routing
 
 Telemetry data will be written to signal specific data streams by default:
-logs to `logs-generic-default`, and traces to `traces-generic-default`.
+logs to `logs-generic-default`, metrics to `metrics-generic-default`, and traces to `traces-generic-default`.
 This can be customised through the following settings:
 
-- `index` (DEPRECATED, please use `logs_index` for logs, `traces_index` for traces): The [index] or [data stream] name to publish events to.
+- `index` (DEPRECATED, please use `logs_index` for logs, `metrics_index` for metrics, `traces_index` for traces): The [index] or [data stream] name to publish events to.
    The default value is `logs-generic-default`.
+
 - `logs_index`: The [index] or [data stream] name to publish events to.  The default value is `logs-generic-default`
-- `logs_dynamic_index` (optional):
-  takes resource or log record attribute named `elasticsearch.index.prefix` and `elasticsearch.index.suffix`
-  resulting dynamically prefixed / suffixed indexing based on `logs_index`. (priority: resource attribute > log record attribute)
+
+- `logs_dynamic_index` (optional): uses resource or log record attributes to dynamically construct index name. See `mode` for details.
   - `enabled`(default=false): Enable/Disable dynamic index for log records
-- `metrics_index`: The [index] or [data stream] name to publish metrics to. The default value is `metrics-generic-default`.
+  - `mode` (default=`prefix_suffix`): defines how dynamic index name is constructed.
```

Comment:
but what if none of these attributes are set? right now, in data_stream mode, we'd fill in the data_stream.* attributes using default values. maybe we would then assume it's also in data_stream mode.

Reply:
Good point. I'd be inclined to do this:
 1. check for `data_stream.*`, use those if set; else
 2. check for `elasticsearch.index.*`, use those if set; else
 3. route to the default data stream and set `data_stream.*` attributes

Reply:
@axw and I talked about this and our current idea is that:
- defaults:
    - logs_dynamic_index.enabled=false
    - traces_dynamic_index.enabled=false
    - metrics_dynamic_index.enabled=true
- there will be no `*_dynamic_index.mode` config to avoid introducing additional config
- setting `*_dynamic_index.enabled`=true means that it will read `data_stream.*` from attributes (the new routing behavior) > read `elasticsearch.index.*` > use default hardcoded `data_stream.*`
- when `*_dynamic_index.enabled`=true, `*_index` (e.g. `metrics_index`) will be ignored, because of the use of hardcoded default `data_stream.*`

@felixbarny wdyt?

Reply:
> when *_dynamic_index.enabled=true, *_index (e.g. metrics_index) will be ignored, because of the use of hardcoded default data_stream.*

wouldn't that break the prefix/suffix mode?

otherwise sgtm

Reply:
> when *_dynamic_index.enabled=true, *_index (e.g. metrics_index) will be ignored, because of the use of hardcoded default data_stream.*

I looked into implementing what I suggested. The above statement is not 100% true. If fields `elasticsearch.index.prefix` and `elasticsearch.index.suffix` exist, we construct the index name from prefix+index+suffix, which may not comply to data stream naming conventions and at that point we should not set `data_stream.*` in the doc. This complicates the design a bit and can be confusing. Now I understand why it was initially designed in that way.

Reply:
How about this then:

- defaults:
    - logs_dynamic_index.enabled=false
    - traces_dynamic_index.enabled=false
    - metrics_dynamic_index.enabled=true
- there will be no *_dynamic_index.mode config to avoid introducing additional config
- setting *_dynamic_index.enabled=true means that it will read data_stream.* from attributes (the new routing behavior) > read elasticsearch.index.* > use default hardcoded data_stream.*
- when *_dynamic_index.enabled=true, *_index (e.g. metrics_index) will be used only in the case where `elasticsearch.index.*` are set

Reply:
See [0567fb1d46d8f61a0491d069bc7641c28ed81ef4](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/33794/commits/0567fb1d46d8f61a0491d069bc7641c28ed81ef4), pending README changes.